File: server_python/password_generator.py
import random

def generate_password(length,options):
    charset = ''
    password = ''
    if (options.letters):
        charset+=options.letters
    elif (options.include_latin_uppercase or options.include_latin_lowercase or options.include_cyrillic_uppercase or options.include_cyrillic_lowercase or options.include_numbers or options.include_symbols):
        if (options.include_latin_uppercase):
            charset += 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
        if (options.include_latin_lowercase):
            charset += 'abcdefghijklmnopqrstuvwxyz'
        if (options.include_cyrillic_uppercase):
            charset += 'АБВГДЕЖЗИЙКЛМНОПРСТУФХЦЧШЩЪЫЬЭЮЯ'
        if (options.include_cyrillic_lowercase):
            charset += 'абвгдежзийклмнопрстуфхцчшщъыьэюя'
        if (options.include_numbers):
            charset += '0123456789'
        if (options.include_symbols):
            charset += '!@#$%^&*()_+-=[]{}|;:,.<>?'
    for _ in range(length):
        password += random.choice(charset)
    return password

File: server_js/passwordGenerator.js
function generatePassword(length, options) {
    let charset = "";
    let password = "";
    
    if (options.letters) charset += options.letters;
    else if (options.includeLatinUppercase || options.includeLatinLowercase || options.includeCyrillicUppercase || options.includeCyrillicLowercase || options.includeNumbers || options.includeSymbols){
        if (options.includeLatinUppercase) charset += 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
        if (options.includeLatinLowercase) charset += 'abcdefghijklmnopqrstuvwxyz';
        if (options.includeCyrillicUppercase) charset += 'АБВГДЕЄЖЗИІЇЙКЛМНОПРСТУФХЦЧШЩЬЮЯ';
        if (options.includeCyrillicLowercase) charset += 'абвгдеєжзиіїйклмнопрстуфхцчшщьюя';
        if (options.includeNumbers) charset += '0123456789';
        if (options.includeSymbols) charset += '/?&<>!@#$%^*()-_=+[]{}|;:,.<>';
    }
    else{
        charset = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
    }

    for (let i = 0; i < length; i++) {
        const randomIndex = Math.floor(Math.random() * charset.length);
        password += charset[randomIndex];
    }
    return password;
}

module.exports = { generatePassword};